contents/javascript/Javascript-framework-Redux.md

```diff
@@ -0,0 +1,226 @@
+<frontmatter>
+  title: Redux
+  header: pagetop.md
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+<div class="website-content">
+  
+{{ booktitle | safe }}
+
+# Introduction to Redux
+
+**Author: [Labayna Neil Brian Narido](https://github.com/nbriannl)** <br>
+Reviewers: 
+
+## What is Redux?
+
+Simply put, Redux is library that manages application state for Javascript applications. You can use it together with view libraries such as React and Vue.
+
+More formally, the [official website](https://redux.js.org/introduction/getting-started) describes Redux as follows:
+
+> *Redux* is a predictable state container for JavaScript apps.
+>
+> It helps you write applications that behave consistently, run in different environments (client, server, and native), and are easy to test. On top of that, it provides a great developer experience, such as live code editing combined with a time traveling debugger.
+>
+> You can use Redux together with React, or with any other view library. It is tiny (2kB, including dependencies), but has a large ecosystem of addons available.
+
+### What is application state?
+
+([Adapted from this Medium article](https://medium.com/javascript-in-plain-english/the-only-introduction-to-redux-and-react-redux-youll-ever-need-8ce5da9e53c6))
+
+Think of it like a global object holding information that you will use for various purposes in your application. 
+
+For example:
+- For a general application you would need to keep store of whether a user is logged in and his/her user information
+- For a to-do list you would need to know what items are currently in the list
+- For a social media application you'd need to store serveral objects and arrays representing the stories to render in the News Feed
+
+All of the above can be considered as state.
+
+Modern web applications have large amount of state. And for complex apps, state is likely to be shared between components and updated from various parts of the code, making state less predictable. Which is why we need libraries like Redux to manage state. 
+
+## How does Redux work?
```

Comment:
Could use a diagram here, e.g. the one you linked in the above [article](https://medium.com/javascript-in-plain-english/the-only-introduction-to-redux-and-react-redux-youll-ever-need-8ce5da9e53c6)

Or perhaps a self made diagram tailored to the earlier example would be nice.
On that note, perhaps the code examples in this section can all follow a central theme ( for e.g. the introductory example ), so the reader can follow better

Reply:
An additional significant sub topic here could be redux middlewares, which are commonly used.

Reply:
Perhaps redux devtools should be moved up to this section here as well, and briefly rehashed in the benefits section, so the reader has better context when reading the benefits

Reply:
I think middleware is too advanced a topic to be put so high up. Maybe as a 'getting started'
Even the redux site put it under advanced 😱  https://redux.js.org/advanced/middleware

Reply:
Good point, maybe you could phrase it as a benefit to replace a more 'advanced' benefit later though.
e.g. `Redux is easily extensible`